File: 075_sort_colors/Solution.java
class Solution {
    public void sortColors(int[] nums) {
        int L = 0, R = nums.length - 1, cur = 0;

        while (cur <= R) {
            if (nums[cur] == 0) {
                swap(nums, cur, L);
                L++; cur++;
            } else if (nums[cur] == 2) {
                swap(nums, cur, R);
                R--;
            } else {
                cur++;
            }
        }
    }

    public void swap(int[] nums, int i, int j) {
        int tmp = nums[i];
        nums[i] = nums[j];
        nums[j] = tmp;
    }

    public static void main(String[] args) {
        int[] nums = {2, 0, 2, 1, 1, 0};
        new Solution().sortColors(nums);
        for (int num : nums)
            System.out.print(num + " ");
    }
}

File: 075_sort_colors/Solution.py
class Solution:
    def sortColors(self, nums: list) -> None:
        l, r, cur = 0, len(nums) - 1, 0

        while cur <= r:
            if nums[cur] == 0:
                nums[cur], nums[l] = nums[l], nums[cur]
                cur += 1
                l += 1
            elif nums[cur] == 2:
                nums[cur], nums[r] = nums[r], nums[cur]
                r -= 1
            else:
                cur += 1


if __name__ == '__main__':
    nums = [2, 0, 2, 1, 1, 0]
    Solution().sortColors(nums)
    print(nums)

